gapis/api/subcmd_idx_trie.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+// SubCmdIdxTrie is a map-based trie using SubCmdIdx for indexing the data
+// stored inside.
+type SubCmdIdxTrie struct {
+	hasValue  bool
```

Comment:
Why do you need `hasValue` and `value`? Are you expecting to have `nil` as a valid value? If not, is there any purpose in the `bool` return value on `GetValue`?